specs/carousel/carousel.md

```diff
@@ -0,0 +1,258 @@
+                role="group"
+                aria-roledescription="slide"
+                aria-label="${x => x.getActiveIndex()} of ${x => x.getItemCount()}"
+            >
+                ${item}
+            </div>`
+            }).join('')
+        }
+        <!-- Solution 2 -->
+        <!-- Have carousel class method that query selects this div and inserts content for the correct item based on id/index from the id's given as an array. -->
+        <div
+            class="carousel-item"
+            role="group"
+            aria-roledescription="slide"
+            aria-label="${x => x.getActiveIndex()} of ${x => x.getItemCount()}"
+        >
+            ${x => x.getActiveItem()}
+        </div>
+    </div>
+    <slot name="previous-flipper"> SEE DEFAULT </slot>
+    <slot name="tab-list">SEE DEFAULT </slot>
+    </div>
+</template>
+```
+
+**Default Flipper**
+```html
+<button
+    @click="${x => x.next()}"
+    class="next-flipper"
+>
+    <svg class="flipper-svg">
+        <path d="example path..."></path>
+    </svg>
+</button>
+```
+
+**Default Play Toggle**
+```html
+<button
+    slot="play-toggle"
+    @click="${x => x.togglePlay()}"
+    class="previous-flipper"
+>
+    <svg class="play-svg">
+        <path d="example path..."></path>
+    </svg>
+</button>
+```
+
+**Default Tab List**
+```html
+<div role="tablist" class="tablist" aria-label="A carousel of items">
+    <div tabindex="0" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+</div>
+```
+
+**Implementation**
+```html
+<body>
+    <fast-carousel
+        // TODO: CHECK is this a way to apply the items?
+        item-0='<img src="...image1.src">'
+        item-1='<img src="...image2.src">'
+        item-2='<img src="...image3.src">'
+    >
+        <button
+            slot="next-flipper"
+            @click="${x => x.next()}" // TODO: How do I attach the carousel.next() function to a slotted item?
+            class="next-flipper"
+        >
+            <svg class="flipper-svg">
+                <path d="example path..."></path>
+            </svg>
+        </button>
+
+        <button
+            slot="previous-flipper"
+            @click="${x => x.next()}"
+            class="previous-flipper"
+        >
+            <svg class="flipper-svg">
+                <path d="example path..."></path>
+            </svg>
+        </button>
+
+        <button
+            slot="play-toggle"
+            @click="${x => x.togglePlay()}"
+            class="previous-flipper"
+        >
+            <svg class="play-svg">
+            <path d="example path..."></path>
+            </svg>
+        </button>
+
+        <div role="tablist" class="tablist" aria-label="A carousel of items">
```

Comment:
I just started looking into a web component version of Tabs might be good to sync up here.